pkg/actions/mustgather.go

```diff
@@ -0,0 +1,364 @@
+}
+
+//writeContainerInspectToFile - writes the results of `docker inspect containerId` to a file
+func writeContainerInspectToFile(containerID, containerName string) error {
+	dockerClient, dockerErr := docker.NewDockerClient()
+	if dockerErr != nil {
+		HandleDockerError(dockerErr)
+		os.Exit(1)
+	}
+	inspectedContents, inspectErr := docker.InspectContainer(dockerClient, containerID)
+	if inspectErr != nil {
+		HandleDockerError(inspectErr)
+		os.Exit(1)
+	}
+	fileContents, _ := json.MarshalIndent(inspectedContents, "", " ")
+	err := ioutil.WriteFile(filepath.Join(mustGatherDirName, containerName+".inspect"), fileContents, 0644)
+	return err
+}
+
+//writeContainerLogToFile - writes the results of `docker logs containerId` to a file
+func writeContainerLogToFile(containerID, containerName string) error {
+	dockerClient, dockerErr := docker.NewDockerClient()
+	if dockerErr != nil {
+		HandleDockerError(dockerErr)
+		os.Exit(1)
+	}
+	logStream, logErr := docker.GetContainerLogs(dockerClient, containerID)
+	if logErr != nil {
+		HandleDockerError(logErr)
+		os.Exit(1)
+	}
+	outFile, createErr := os.Create(filepath.Join(mustGatherDirName, containerName+".log"))
+	if createErr != nil {
+		errors.CheckErr(createErr, 201, "")
+	}
+	defer outFile.Close()
+	_, err := io.Copy(outFile, logStream)
+	return err
+}
+
+//copyCodewindWorkspace - copies the Codewind PFE container's workspace to mustgather
+func copyCodewindWorkspace(containerID string) error {
+	dockerClient, dockerErr := docker.NewDockerClient()
+	if dockerErr != nil {
+		HandleDockerError(dockerErr)
+		os.Exit(1)
+	}
+	tarFileStream, fileErr := docker.GetFilesFromContainer(dockerClient, containerID, "/codewind-workspace")
+	if fileErr != nil {
+		HandleDockerError(fileErr)
+		os.Exit(1)
+	}
+	defer tarFileStream.Close()
+	// Extracting tarred files
+	tarBallReader := tar.NewReader(tarFileStream)
+
+	for {
+		header, err := tarBallReader.Next()
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+			return (err)
+		}
+
+		// get the individual filename and extract to the current directory
+		filename := header.Name
+
+		switch header.Typeflag {
+		case tar.TypeDir:
+			// handle directory
+			err = os.MkdirAll(filepath.Join(mustGatherDirName, filename), os.FileMode(header.Mode)) // or use 0755 if you prefer
+			if err != nil {
+				return err
+			}
+
+		case tar.TypeReg:
+			// handle normal file
+			writer, err := os.Create(filepath.Join(mustGatherDirName, filename))
+			if err != nil {
+				return err
+			}
+
+			io.Copy(writer, tarBallReader)
+
+			err = os.Chmod(filepath.Join(mustGatherDirName, filename), os.FileMode(header.Mode))
+			if err != nil {
+				return err
+			}
+
+			writer.Close()
+		default:
+			logMG("Unable to untar type : " + string(header.Typeflag) + " in file " + filename)
+		}
+	}
+	return nil
+}
+
+//copyFileHere - copies the contents of the source file to a target file in the mustgather directory
+func copyFileHere(sourceFilePath, targetFile string) error {
```

Comment:
`copyFileHere` - this takes a location so `Here` is redundant. Also could this move to pkg/utils/filesystem.go?

Reply:
Agreed - moved to pkg/utils/filesystem.go/CopyFile